app/models/patient.rb

```diff
@@ -48,6 +48,31 @@ class Patient < ApplicationRecord
 
   validates :assigned_user, numericality: { only_integer: true, allow_nil: true, greater_than: 0, less_than_or_equal_to: 9999 }
 
+  validates :last_date_of_exposure, earliest_date: { date: Date.new(2020, 1, 1) },
+                                    latest_date: { date: 30.days.from_now.to_date },
```

Comment:
Is there anything we need to consider with Timezone here? If someone with local time ET sets a last_date_of_exposure for today + 30 days, but it's already tomorrow in UTC time (which is the timezone on our prod instance), the user entered date will be considered invalid here even though technically it's valid for them, right? I think this again speaks to the need to make sure ALL dates entered by the user are converted to UTC on the backend, and displayed in their local timezone on the frontend because we keep running into stuff like this.

Reply:
> If someone with local time ET sets a last_date_of_exposure for today + 30 days, but it's already tomorrow in UTC time (which is the timezone on our prod instance), the user entered date will be considered invalid here even though technically it's valid for them, right?

Today is Oct 16 in ET and Oct 17 in UTC
The validation states that `last_date_of_exposure` is within 30 days.
Oct 16 + 30 days < Oct 17 + 30 days so the date is valid.

So your example is not really where the problem arises. It's when the user is submitting a time that has happened before UTC.

Let's assume that today is Oct 18 just West of the International Date Line (UTC + 14) and Oct 17 UTC
The validation states that `last_date_of_exposure` is within 30 days.
Oct 18 + 30 days > Oct 17 + 30 days so the date is invalid.

What can we do about in this case? We only pass strings from the frontend to the backend. Strings contain no TZ information which is required to correctly convert from local time to UTC. We will need to convert input to provide full DateTime information.